custom_components/localtuya/core/pytuya/__init__.py

```diff
@@ -1327,6 +1327,14 @@ def _decode_payload(self, payload):
 
         # We will store the payload to trigger an event in HA.
         if "dps" in json_payload:
+            ts = json_payload.get("t")
```

Comment:
Could you add a link to your discussion - https://github.com/xZetsubou/hass-localtuya/discussions/507#discussioncomment-13032485 - as a comment around why this wrapping behaviour is done?

It didn't make immediate sense until reading that (I found myself asking "are energy monitoring DPs the only ones with timestamps?" "Should this structure emit the same thing with a timestamp of None for other areas?" "What does the structure look like")

Reply:
done :)